### ISSUE TITLE
[Publisher] Updated designs for Agency Settings page

### ISSUE THREAD

**Issue by lilidworkin** [Recidiviz/justice-counts#244]
Update the designs of the Agency Settings page according to [this Figma](https://www.figma.com/file/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?node-id=6255%3A15227&t=toeuAepUpEuJakyI-0).

**Basic Information**
Only the styling of this section has changed.

**Agency Purpose and Functions**
This section has been renamed to "Agency Information". It also now has two different states: a read-only state and an edit state. When you click "Edit description", you move to the edit state. The functionality should remain the same (i.e. use the same backend API should be used to get/set the text).

**Supervision Populations**
This section now has a read-only state and an edit state. The functionality should remain the same (i.e. use the same backend API should be used to get/set the text).

**Team Management**
This section is now editable (before it was read-only). The backend API for this feature is not ready yet, unfortunately, so just do as much of the frontend work as you can.

**Jurisdictions**
This design for this section is finished and is now ready to be implemented. The backend API for this feature is also not ready yet, so just do as much of the frontend work as you can. _I'm curious if @ivan-kishko , @terryttsai , or @mxosman have thoughts on how to best implement the typeahead searchbox_. Should the backend handle this, and the frontend would just repeatedly hit a backend API every time the user types a new character, and get a list of matches in response? Or should the frontend store a list of all possible jurisdictions and manage the autocomplete functionality?

**Comment by ivan-kishko** [comment]
Hello. Whenever you see it, have a question - can more than 1 section be in edit mode or only one at a time?

**Comment by mxosman** [comment]
Regarding the Jurisdiction's question: I think we can have have a file on the frontend that houses all of the jurisdictions (since they are mostly static) - we'll avoid an API call all together this way. I'm also good with Ivan's proposal if we want to centralize things in the BE. Either option could work! Storing it on the frontend feels like the quickest snappiest option in terms of UX.

**Comment by hobuobi** [comment]
Great questions! Let me try to answer these as best as I can: 

> Can more than 1 section be in edit mode or only one at a time?
No. If they try to edit another section without closing the first, the screen should move back to the open editMode section. If you want to have have fun with it, I would also animate the open section somehow (make the section flash quickly three times. shake, etc). 


> Cannot see sub menu on new design, should I remove it?
Remove it.


> Team management - there are 2 buttons in edit mode Done and Cancel - even though backend is not available seems like cancel and done buttons doing the same (closing edit mode) opposed for example Save button for supervisions of description which is exactly saving changes, so maybe there should be only "Done" button?
The difference is that "Cancel" would close edit mode without saving the changes, while "Done" would save the changes and then close edit mode. This is different from the current behavior, which I believe saves the settings with every change. 
I do think a different word should be used, like "Save Changes" or something. 


> Team management - when you delete user there is a confirmation - is it modal (and if so is it modal for section or page) or it should just replace team management editMode section?
I'm not actually sure what @agronomic 's intention was here. For now, if you need a decision on it, I would replace all the content in the editMode section with the modal content. We can confirm with Juan when they're back. 


> Team management - how should it look in non edit mode - email to right or email below - there are 2 screens without edit mode for team and they look different
Email to right. 


> Jurisdictions - maybe instead of hardcoding on front or hitting api with each letter we can make a static list on a backend with all possible jurisdictions and get it from backend with api call on frontend to work with?
Agreed with Mahmoud; saving on the frontend seems like the way to go. 

> Not sure what is going on with exclusions in jurisdictions - is there like static set of cities/states/counties and you can exclude/include only them? also you can exclude either by checkbox or input? and what should "Add them" in bottom left corner do?
A few comments: 
- Basically, a user can identify both "Included Areas" and "Excluded Areas." In both cases, the user is just appending census areas to a list. 
- Since exclusions are not common and generally discouraged, the user must click the "Add them" link in the bottom left to enter the flow for exclusions. This flow should be very similar to the one for Included Areas (the default). 
- However, I agree that the mocks are not clear about how this separate flow fits into the rest of the picture... and I did not make them, so I unfortunately can't provide all the answers you need. I would proceed without this for now, and then we can address it when Juan is back next Tuesday. I have an idea of what this should look like, so I can also mock that up if need be.

**Comment by ivan-kishko** [comment]
Hello @hobuobi! Thanks for answers. I think though that you didnt get what I meant to ask about team management buttons. Thing is, From design it feels like particularly in team management all actions are instant and do not have to be saved explicitly like in supervision or description. If you click "Invite" looks like invite should be sent right away, same with make admin and remove user, especially remove user - if we do not delete user after confirmation in modal and instead in addition after that have to hit "done/save" to actually remove one that looks weird. So taking all the above feels like here have to be only one button which will just close setting because all actions inside team management are instant. I guess we have to wait all guys to be sure what is going on there

**Comment by lilidworkin** [comment]
As always you are asking excellent questions @ivan-kishko ! Just going to weigh in quickly on a couple of things:

> Jurisdictions - maybe instead of hardcoding on front or hitting api with each letter we can make a static list on a backend with all possible jurisdictions and get it from backend with api call on frontend to work with?

While @mxosman and @hobuobi are right that the list of jurisdictions is static and could be stored on the frontend, I'd like to avoid doing that, because we'll have to store them on the backend as well, and having data duplication could cause bugs. @ivan-kishko I think you're idea is exactly right -- the backend should send over the list of jurisdictions (either as part of an existing GET request or a new one) and then the frontend can save the list in memory, so that the typeahead is still fast and snappy. This seems like the best of both worlds. I'll figure out the structure of the response and get back to you ASAP so you can mock it out during development. I imagine it will be a list of objects that looks something like this: 

```
[
  {
     "type": "county",
    "name": "Thurston County",
    "state": "WA",
   },
  {
     "type": "city",
    "name": "Plano",
    "state": "CA",
  },
]
```

> From design it feels like particularly in team management all actions are instant and do not have to be saved explicitly like in supervision or description. 

@ivan-kishko  I see what you're saying and you're right -- from the designs, it certainly seems like inviting/removing users should be instant and should happen right away, rather than waiting for the user to press "Done". @hobuobi does that make sense to you too? In that case I think we can remove the "Cancel" button and just have the "Done" button (which wouldn't do anything and would just close the modal.
